native/src/hooks/useLoadCities.ts

```diff
@@ -1,24 +1,43 @@
 import { useCallback } from 'react'
 
-import { CityModel, createCitiesEndpoint, ReturnType, useLoadAsync } from 'api-client'
+import { CityModel, createCitiesEndpoint, fromError, ReturnType, useLoadAsync } from 'api-client'
 
+import { SnackbarType } from '../components/SnackbarContainer'
 import dataContainer from '../utils/DefaultDataContainer'
 import { determineApiUrl } from '../utils/helpers'
 
-const useLoadCities = (): ReturnType<CityModel[]> => {
-  const load = useCallback(async () => {
-    if (await dataContainer.citiesAvailable()) {
-      return dataContainer.getCities()
-    }
+type Params = {
+  forceUpdate?: boolean
+  showSnackbar: (snackbar: SnackbarType) => void
+}
+
+const loadWithCache = async ({ showSnackbar, forceUpdate = false }: Params): Promise<CityModel[] | null> => {
+  const cachedData = (await dataContainer.citiesAvailable()) ? await dataContainer.getCities() : null
+
+  if (!forceUpdate && cachedData) {
+    return cachedData
+  }
 
+  try {
     const payload = await createCitiesEndpoint(await determineApiUrl()).request()
     if (payload.data) {
       await dataContainer.setCities(payload.data)
     }
-    return payload.data ?? null
-  }, [])
-
-  return useLoadAsync(load)
+    return payload.data ?? cachedData
+  } catch (e) {
+    if (!cachedData) {
+      throw e
+    }
+    if (forceUpdate) {
+      showSnackbar({ text: fromError(e) })
+    }
+  }
+  return cachedData
 }
 
+const useLoadCities = (params: Params): ReturnType<CityModel[]> =>
+  // Normally using params as dependency triggers infinite re-renders
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  useLoadAsync(useCallback(forceUpdate => loadWithCache({ ...params, forceUpdate }), []))
```

Comment:
This should only be true if the params are not stateful. 
When I tested using only `useLoadCities(showSnackbar)` there were no rerenders anymore. Aparrently `{ showSnackbar } creates a new object which is not stateful and therefore triggers those re-renders. Please change this
```suggestion
const useLoadCities = (showSnackbar: (snackbar: SnackbarType) => void): ReturnType<CityModel[]> =>
  useLoadAsync(useCallback(forceUpdate => loadWithCache({ showSnackbar, forceUpdate }), [showSnackbar]))
```

Reply:
Good catch, guess I just copied it haha :)